### PR TITLE
fix: sort by name, model provider & filter for agent's model dropdown

### PR DESCRIPTION
**Pull request by ivyjeong13**
* Addresses #838 

update: added checkmark for selected to match with select
Screenshot 2024-12-16 at 5 39 27 PM

https://github.com/user-attachments/assets/a10c98ad-a680-4ca9-8979-caa1ddd9dc6f

Screenshot 2024-12-16 at 4 31 17 PM

* add `ComboBox` for dropdown with filter -- also comes with mobile responsive drawer!

### CODE REVIEW

File: ui/admin/app/components/ui/drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+"use client";
+
+import * as React from "react";
+import { Drawer as DrawerPrimitive } from "vaul";
+
+import { cn } from "~/lib/utils";
+
+const Drawer = ({
```
Comment (ivyjeong13) [inline ui/admin/app/components/ui/drawer.tsx:6]: Pulled Combobox from shadcn example, came with a mobile responsive but can strip it out if needed!

**Comment by ivyjeong13** [comment]
If we're alright with this, I'll update the default models dropdowns to use this as well!

File: ui/admin/app/components/composed/ComboBox.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
+"use client";
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/composed/ComboBox.tsx]: nit: can you remove the "use client" directives?

File: ui/admin/app/components/ui/drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
+"use client";
```
Comment (ryanhopperlowe) [inline ui/admin/app/components/ui/drawer.tsx]: same here

**Review by ryanhopperlowe — APPROVED** [review]

**Pull request merged**